File: nova-gestion-backend/src/main/java/nova/gestion/model/WorkSheet.java
package nova.gestion.model;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.sql.Date;
import java.util.ArrayList;

@Data
@AllArgsConstructor
public class WorkSheet {
    private int idWorkSheet;
    private Status status;
    private int quantity;
    private Date dateCreation;
    private Date dueDate;
    private String orderNumber;
    private ArrayList<Task> tasks;

    public WorkSheet(int idWorkSheet, int quantity, Date dateCreation, Date dueDate, String orderNumber)
    {
        this.idWorkSheet = idWorkSheet;
        this.quantity = quantity;
        this.dateCreation = dateCreation;
        this.dueDate = dueDate;
        this.orderNumber = orderNumber;
    }
}

File: webui/src/app/models/worksheet.ts
import {Client} from './client';
import {Status} from './status';
import {Program} from './program.model';
import {Task} from './task';

export class Worksheet {
  public idWorkSheet: number;
  public quantity: number;
  public orderNumber: string;
  public dueDate: Date;
  public dateCreation: Date;
  public client: Client;
  public status: Status;
  public program: Program;
  public tasks: Task[];
}
